### PR TITLE
Add prescription-sync cronjob

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Related: https://github.com/thoth-station/thoth-application/issues/1052

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: prescription-sync-job/base/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  name: prescription-sync
+  labels:
+    app: thoth
+    component: prescription-sync
+spec:
+  schedule: "@daily"
+  suspend: true
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 2
+  failedJobsHistoryLimit: 4
```
Comment (harshad16) [inline prescription-sync-job/base/cronjob.yaml:14]:
```suggestion
  successfulJobsHistoryLimit: 1
  failedJobsHistoryLimit: 2
```
I think 2 failed jobs history is good enough.

File: prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'I have successfully synchronized *prescription-sync* components to *OCP4-STAGE*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-prescription-sync|ArgoCD UI>'}"
```
Comment (harshad16) [inline prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml:22]:
```suggestion
            - "{'text':'I have successfully synchronized *prescription-sync* components to *OCP4-STAGE*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/stage-thoth-prescription-sync|ArgoCD UI>'}"
```

File: prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        - name: chat-notification
+          image: registry.access.redhat.com/ubi8/ubi
+          command:
+            - "curl"
+            - "-X"
+            - "POST"
+            - "-H"
+            - "Content-Type: application/json; charset=UTF-8"
+            - "-d"
+            - "{'text':'🔥 *FAIL* synchronizing *prescription-sync* components to *OCP4-STAGE*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/test-thoth-prescription-sync|ArgoCD UI>'}"
```
Comment (harshad16) [inline prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml:53]:
```suggestion
            - "{'text':'🔥 *FAIL* synchronizing *prescription-sync* components to *OCP4-STAGE*, see <https://argocd-server-aicoe-argocd.apps.ocp4.prod.psi.redhat.com/applications/stage-thoth-prescription-sync|ArgoCD UI>'}"
```

File: prescription-sync-job/overlays/prod/cronjob.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (harshad16) [inline prescription-sync-job/overlays/prod/cronjob.yaml:1]: we can remove the `prod` overlays, we are no longer maintaining them.

File: prescription-sync-job/overlays/test/imagestreamtag.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: prescription-sync-job
+spec:
+  tags:
+    - name: latest
+      from:
+        kind: DockerImage
+        name: quay.io/thoth-station/prescription-sync-job:v0.8.1
```
Comment (harshad16) [inline prescription-sync-job/overlays/test/imagestreamtag.yaml:10]:
```suggestion
        name: quay.io/thoth-station/prescription-sync-job:v0.0.1
```

**Review by harshad16 — COMMENTED** [review]
lgtm, thanks
left we comments :)

File: prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml (hunk 2 above)
Comment (fridex) [reply prescription-sync-job/overlays/ocp4-stage/thoth-notification.yaml:22]: I copied templates from graph-backup, they might require similar changes as I kept this part untouched.

File: prescription-sync-job/overlays/prod/cronjob.yaml (hunk 4 above)
Comment (fridex) [reply prescription-sync-job/overlays/prod/cronjob.yaml]: Removed in 339dcf4 :+1:

**Review by harshad16 — APPROVED** [review]
thanks :100:

**Pull request merged**